compiler/src/iree/compiler/Dialect/VM/Conversion/StandardToVM/ConvertStandardToVM.cpp

```diff
@@ -1085,6 +1085,80 @@ class CondBranchOpConversion : public OpConversionPattern<cf::CondBranchOp> {
   }
 };
 
+class SwitchOpConversion : public OpConversionPattern<cf::SwitchOp> {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(cf::SwitchOp srcOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    // Special handling for default only ops: just jump to default.
+    if (srcOp.getCaseDestinations().empty()) {
+      rewriter.replaceOpWithNewOp<IREE::VM::BranchOp>(
+          srcOp, srcOp.getDefaultDestination(), adaptor.getDefaultOperands());
+      return success();
+    }
+
+    // NOTE: cf.switch can have sparse indices but we cannot; instead we fill
+    // any gaps with branches to the default block. This is wasteful but keeps
+    // the runtime super simple - if we have offset or really sparse tables
+    // (default + case 10000 + case 400000) we can optimize those in the
+    // compiler by using multiple branch tables, inverse lookups via a lookup
+    // op, etc.
```

Comment:
Are you aware of any usage of `cf.switch` / `scf.index_switch` that existing frameworks pass through to IREE? This seems fine if not, but it is a bit of a landmine :P

I wonder if we could get away with an assert (or warning remark?) here, checking for values > 1000.

Reply:
Nothing today as things would have exploded, but yeah definitely something we'll want to watch for. I doubt many people are using scf.index_switch because the lowering to cf.switch casts the index (the namesake of index_switch) to i32, meaning that it's really scf.i32_switch lol. Works for us because we want an i32 case but implies that people aren't using this as a dictionary replacement or something where the cases are hash codes.